src/components/TaskDetail.jsx

```diff
@@ -104,192 +106,276 @@ export function TaskDetail({ task, isOpen, onClose, onUpdate, onDelete, onAddLab
+                  <div className="editor-actions">
+                    <button 
+                      className="save-button"
+                      onClick={handleSave}
                     >
-                      ×
+                      Save
                     </button>
-                  )}
+                  </div>
+                </div>
+              ) : (
+                <div className="description-preview" data-color-mode={isDark ? "dark" : "light"}>
+                  <MDEditor.Markdown 
+                    source={description || 'No description provided.'} 
+                    rehypePlugins={[[rehypeSanitize]]}
+                    style={{
+                      backgroundColor: 'transparent',
+                      color: 'inherit'
+                    }}
+                  />
                 </div>
-              ))}
-              {(!task.labels || task.labels.length === 0) && (
-                <div className="no-labels-message">
-                  No labels added yet
+              )}
+            </div>
+
+            {/* Comments Section */}
+            <div className="task-comments">
+              <div className="comments-header">
+                <h3>Comments</h3>
+              </div>
+              
+              {user && (
+                <div className="comment-form">
+                  <textarea
+                    value={newComment}
+                    onChange={(e) => setNewComment(e.target.value)}
+                    placeholder="Write a comment..."
+                    className="comment-input"
+                    rows={3}
+                  />
+                  <button 
+                    className="add-comment-button"
+                    onClick={handleAddComment}
+                    disabled={!newComment.trim()}
+                  >
+                    Add Comment
+                  </button>
                 </div>
               )}
+
+              <div className="comments-list">
+                {(task.comments || []).length > 0 ? (
+                  task.comments.map(comment => (
+                    <div key={comment.id} className="comment-item">
+                      <div className="comment-header">
+                        <span className="comment-author">{comment.createdByName}</span>
+                        <span className="comment-date">
+                          {new Date(comment.createdAt).toLocaleString()}
+                        </span>
+                        {user && user.uid === comment.createdBy && (
+                          <button
+                            className="delete-comment-button"
+                            onClick={() => handleDeleteComment(comment.id)}
+                            title="Delete comment"
+                          >
+                            ×
+                          </button>
+                        )}
+                      </div>
+                      <div className="comment-text">{comment.text}</div>
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
import DOMPurify from 'dompurify';

// In the render section:
<div className="comment-text">
  {DOMPurify.sanitize(comment.text)}
</div>
```
The comment text should be sanitized before being displayed to prevent XSS attacks, similar to how the description is handled with rehypeSanitize.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;